Diff with geth upstream master

selevit opened the pull request.

soc1c commented:
Good catch. Can you check again?

meowsbits commented:
You mean just the peer name like below?

```
> admin.addPeer("enode://2d95d69f8d212a3eef132dd615b3b7bf41dc01dd4703185c9a12a5814c57c35a9ff406b9d353f005333ea3284c63732534e2c55e4619cfb5c4f872bb453e80e6@127.0.0.1:30303?discport=0");
true
> admin.peers
[{
    caps: ["eth/63", "eth/64"],
    enode: "enode://2d95d69f8d212a3eef132dd615b3b7bf41dc01dd4703185c9a12a5814c57c35a9ff406b9d353f005333ea3284c63732534e2c55e4619cfb5c4f872bb453e80e6@127.0.0.1:30303",
    id: "0de87a6f64aebc03e7309741fd6a88f1c77ff93c1ce6a5ecf06bda6bf8eeffa7",
    name: "CoreGeth/v1.9.10-unstable-21f13900-20200218/linux-amd64/go1.13.5",
    network: {
      inbound: false,
      localAddress: "127.0.0.1:42344",
      remoteAddress: "127.0.0.1:30303",
      static: true,
      trusted: false
    },
    protocols: {
      eth: {
        difficulty: 17179869184,
        head: "0xd4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3",
        version: 64
      }
    }
}]
```

soc1c commented:
Yeah, looks good.

meowsbits commented:
Looks like we need to clean up Appveyor, out-of-storage-space error is causing the builds to fail.

![2020-02-18-171828_1017x382_screenshot](https://user-images.githubusercontent.com/45600330/74783029-f8575600-5261-11ea-966c-5a62f09cced3.png)
https://ci.appveyor.com/project/chunfuyang/multi-geth-5su26/builds/30892853

meowsbits commented:
Ha, although the Appveyor retention policy issue may just go away when we swap namespaces.

gballet committed:
trie: quell linter in commiter.go (#21329)

holiman committed:
core/vm: less allocations for various call variants  (#21222)

* core/vm/runtime/tests: add more benchmarks

* core/vm: initial work on improving alloc count for calls to precompiles

name                                  old time/op    new time/op    delta
SimpleLoop/identity-precompile-10M-6     117ms ±75%      43ms ± 1%  -63.09%  (p=0.008 n=5+5)
SimpleLoop/loop-10M-6                   79.6ms ± 4%    70.5ms ± 1%  -11.42%  (p=0.008 n=5+5)

name                                  old alloc/op   new alloc/op   delta
SimpleLoop/identity-precompile-10M-6    24.4MB ± 0%     4.9MB ± 0%  -79.94%  (p=0.008 n=5+5)
SimpleLoop/loop-10M-6                   13.2kB ± 0%    13.2kB ± 0%     ~     (p=0.357 n=5+5)

name                                  old allocs/op  new allocs/op  delta
SimpleLoop/identity-precompile-10M-6      382k ± 0%      153k ± 0%  -59.99%  (p=0.000 n=5+4)
SimpleLoop/loop-10M-6                     40.0 ± 0%      40.0 ± 0%     ~     (all equal)

* core/vm: don't allocate big.int for touch

name                                  old time/op    new time/op    delta
SimpleLoop/identity-precompile-10M-6    43.3ms ± 1%    42.4ms ± 7%     ~     (p=0.151 n=5+5)
SimpleLoop/loop-10M-6                   70.5ms ± 1%    76.7ms ± 1%   +8.67%  (p=0.008 n=5+5)

name                                  old alloc/op   new alloc/op   delta
SimpleLoop/identity-precompile-10M-6    4.90MB ± 0%    2.46MB ± 0%  -49.83%  (p=0.008 n=5+5)
SimpleLoop/loop-10M-6                   13.2kB ± 0%    13.2kB ± 1%     ~     (p=0.571 n=5+5)

name                                  old allocs/op  new allocs/op  delta
SimpleLoop/identity-precompile-10M-6      153k ± 0%       76k ± 0%  -49.98%  (p=0.029 n=4+4)
SimpleLoop/loop-10M-6                     40.0 ± 0%      40.0 ± 0%     ~     (all equal)

* core/vm: reduce allocs in staticcall

name                                  old time/op    new time/op    delta
SimpleLoop/identity-precompile-10M-6    42.4ms ± 7%    37.5ms ± 6%  -11.68%  (p=0.008 n=5+5)
SimpleLoop/loop-10M-6                   76.7ms ± 1%    69.1ms ± 1%   -9.82%  (p=0.008 n=5+5)

name                                  old alloc/op   new alloc/op   delta
SimpleLoop/identity-precompile-10M-6    2.46MB ± 0%    0.02MB ± 0%  -99.35%  (p=0.008 n=5+5)
SimpleLoop/loop-10M-6                   13.2kB ± 1%    13.2kB ± 0%     ~     (p=0.143 n=5+5)

name                                  old allocs/op  new allocs/op  delta
SimpleLoop/identity-precompile-10M-6     76.4k ± 0%      0.1k ± 0%     ~     (p=0.079 n=4+5)
SimpleLoop/loop-10M-6                     40.0 ± 0%      40.0 ± 0%     ~     (all equal)

* trie: better use of hasher keccakState

* core/state/statedb: reduce allocations in getDeletedStateObject

* core/vm: reduce allocations in all call derivates

* core/vm: reduce allocations in call variants

- Make returnstack `uint32`
- Use a `sync.Pool` of `stack`s

* core/vm: fix tests

* core/vm: goimports

* core/vm: tracer fix + staticcall gas fix

* core/vm: add back snapshot to staticcall

* core/vm: review concerns + make returnstack pooled + enable returndata in traces

* core/vm: fix some test tracer method signatures

* core/vm: run gencodec, minor comment polish

Co-authored-by: Péter Szilágyi <peterke@gmail.com>

madjarevicn committed:
core: raise gas limit in --dev mode, seed blake precompile (#21323)

* Set gasLimit in --dev mode to be 9m.

* core: Set gasLimit to 11.5 milion and add 1 wei allocation for BLAKE2b

karalabe committed:
core/vm: use pointers to operations vs. copy by value

karalabe committed:
ethstats: fix reconnection issue, implement primus pings

karalabe committed:
Merge pull request #21347 from karalabe/ethstats-fixes

ethstats: fix reconnection issue, implement primus pings

rjl493456442 committed:
accounts, internal: fix funding check when estimating gas (#21346)

* internal, accounts: fix funding check when estimate gas

* accounts, internal: address comments

karalabe committed:
params: release Geth v1.9.17

karalabe committed:
params: begin v1.9.18 release cycle

karalabe committed:
Merge pull request #21336 from karalabe/tiny-ref-optimization

core/vm: use pointers to operations vs. copy by value

binacs committed:
les: make clientPool.connectedBias configurable (#21305)

sammy007 committed:
ethclient: serialize negative block number as "pending" (#21177)

Fixes #21175

Co-authored-by: sammy007 <sammy007@users.noreply.github.com>
Co-authored-by: Adam Schmideg <adamschmideg@users.noreply.github.com>

renaynay committed:
whisper/whisperv6: improve test error messages (#21348)
